groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -0,0 +1,1653 @@
+    void emplace(bslma::Allocator              *allocator,
+                 std::initializer_list<IL_TYPE> initializer_list,
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_01),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_02),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_03),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_04),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_05),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_06),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_07),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_08),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_09),
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS_10));
+#endif  // BSLSTL_OPTIONAL_VARIADIC_LIMIT_A >= 10
+
+
+    #endif
+#else
+// The generated code below is a workaround for the absence of perfect
+// forwarding in some compilers.
+    template<class... ARGS>
+    void emplace(bslma::Allocator *allocator,
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)...);
+
+    #if defined(BSLS_COMPILERFEATURES_SUPPORT_GENERALIZED_INITIALIZERS)
+    template<class IL_TYPE, class... ARGS>
+    void emplace(bslma::Allocator              *allocator,
+                 std::initializer_list<IL_TYPE> initializer_list,
+                 BSLS_COMPILERFEATURES_FORWARD_REF(ARGS)...);
+
+    #endif
+// }}} END GENERATED CODE
+#endif
+
+    void reset() BSLS_KEYWORD_NOEXCEPT;
+        // Destroy the 'value_type' object in 'd_buffer', if any.
+
+    TYPE& value() BSLS_KEYWORD_LVREF_QUAL;
+        // Return the 'value_type' object in 'd_buffer' with const
+        // qualification adjusted to match that of 'TYPE'.  The behavior is
+        // undefined unless 'this->hasValue() == true'
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS)
+    TYPE&& value() BSLS_KEYWORD_RVREF_QUAL;
+        // Return the 'value_type' object in 'd_buffer' with const
+        // qualification adjusted to match that of 'TYPE'.  The behavior is
+        // undefined unless 'this->hasValue() == true'
+#endif // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+    //ACCESSORS
+    bool hasValue() const BSLS_KEYWORD_NOEXCEPT;
+        // return 'true' if there is a value in 'd_buffer', and 'false'
+        //otherwise.
+
+    const TYPE& value() const BSLS_KEYWORD_LVREF_QUAL;
+        // Return the 'value_type' object in 'd_buffer' with const
+        // qualification adjusted to match that of 'TYPE'.  The behavior is
+        // undefined unless 'this->hasValue() == true'
+
+#if defined(BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS)
+    const TYPE&& value() const BSLS_KEYWORD_RVREF_QUAL;
+        // Return the 'value_type' object in 'd_buffer' with const
+        // qualification adjusted to match that of 'TYPE'.  The behavior is
+        // undefined unless 'this->hasValue() == true'
+#endif // BSLS_COMPILERFEATURES_SUPPORT_REF_QUALIFIERS
+
+};
+
+template <class TYPE>
```

Comment:
all inline definitions moved to the end of the file (Except the nullopt_t::nullopt_t(int) because we need that definition before the nullopt variable), in the order they are declared in, and with the same corresponding comments. I am not closing this conversation as I think it would be a good idea to check these changes after I also run the bde_format script.

Reply:
Did you have any success integrating bde-format with the editor?

Reply:
I found that running the bde-format tool in a terminal and looking at the differences in the eclipse compare tool is a very simple way to pick and choose what gets formatted (provided I know which bits to skip, of course). Consequently, I haven't bothered with the integration.